astropy/table/jsviewer.py

```diff
@@ -84,12 +84,19 @@ class Conf(_config.ConfigNamespace):
 
 
 HTML_JS_SCRIPT = """
+jQuery.extend( jQuery.fn.dataTableExt.oSort, {{
+    "optionalnum-pre": function (a) {{ var b=parseFloat(a); if (isNaN(b)) return Number.NEGATIVE_INFINITY; else return b;}},
+    "optionalnum-asc": function (a,b) {{ return ((a < b) ? -1 : ((a > b) ? 1 : 0)); }},
+    "optionalnum-desc": function (a,b) {{ return ((a < b) ? 1 : ((a > b) ? -1 : 0)); }}
+}});
+
 $(document).ready(function() {{
     $('#{tid}').dataTable({{
      "iDisplayLength": {display_length},
      "aLengthMenu": {display_length_menu},
      "bJQueryUI": true,
-     "sPaginationType": "full_numbers"
+     "sPaginationType": "full_numbers",
+     "aoColumnDefs": [{{"sType": "optionalnum", "aTargets": {sort_columns}}}]
```

Comment:
Ah, you're right.  I knew there was code in here specifically for displaying an interactive table in IPython Notebook.  But you're right that it's no longer accessible through the user API, and I'm not sure whether it worked in the first place.  I added more details here: https://github.com/astropy/astropy/issues/3937#issuecomment-131940639

In short I think it should be considered dead code, so you can safely ignore it for the purpose of this PR.